sdk/src/transaction.rs

```diff
@@ -503,32 +355,31 @@ mod tests {
         assert_eq!(tx, deser);
     }
 
+    /// Detect changes to the serialized size of payment transactions, which affects TPS.
     #[test]
     fn test_transaction_serialized_size() {
-        let keypair = Keypair::new();
-        let program_id = Pubkey::new(&[4; 32]);
-        let to = Pubkey::new(&[5; 32]);
-        let tx = Transaction::new_signed(
-            &keypair,
-            &[keypair.pubkey(), to],
-            &program_id,
-            &(1u8, 2u8, 3u8),
-            Hash::default(),
-            99,
+        let alice_keypair = Keypair::new();
+        let alice_pubkey = alice_keypair.pubkey();
+        let bob_pubkey = Keypair::new().pubkey();
+        let ix = SystemInstruction::new_move(&alice_pubkey, &bob_pubkey, 42);
+        let tx =
+            Transaction::new_signed_instructions(&[&alice_keypair], vec![ix], Hash::default(), 0);
+
+        // Note: With the shortvec optimization, this should drop to 13 bytes.
+        // program_ids_index=1, accounts=1+1, data=1+9
+        assert_eq!(
+            serialized_size(&tx.instructions[0]).unwrap(),
+            31,
+            "unexpected serialized instruction size"
+        );
+
+        // Note: With the shortvec optimization, this should drop to 217 bytes.
+        // signatures=1+64, account_keys=1+64, recent_blockhash=32, fee=8, program_ids=1+32, instructions=1+13
```

Comment:
@sakridge, can you check my math here?

Reply:
This is wrong. Patch coming soon. The enum's tag is 4 bytes, not 1.  Also, accounts should be 1+2. Minimum sizes should b 17 and 221.  @sakridge, no need for review here after all - got it implemented locally.